src/sql/workbench/services/restore/browser/restoreDialog.ts

```diff
@@ -170,7 +185,31 @@ export class RestoreDialog extends Modal {
 	protected renderBody(container: HTMLElement) {
 		const restoreFromElement = DOM.$('.restore-from');
 		this.createLabelElement(restoreFromElement, localize('source', "Source"), true);
-		this._restoreFromSelectBox = this.createSelectBoxHelper(restoreFromElement, localize('restoreFrom', "Restore from"), [this._databaseTitle, this._backupFileTitle], this._databaseTitle);
+		this._restoreFromSelectBox = this.createSelectBoxHelper(restoreFromElement, localize('restoreFrom', "Restore from"), [this._databaseTitle, this._backupFileTitle, this._urlTitle], this._databaseTitle);
+
+		this._restoreFromUrlElement = DOM.$('.backup-url');
+		DOM.hide(this._restoreFromUrlElement);
+		const urlErrorMessage = localize('missingBackupUrlError', "Backup url is required.");
+		const urlValidationOptions: IInputOptions = {
+			validationOptions: {
+				validation: (value: string) => !value ? ({ type: MessageType.ERROR, content: urlErrorMessage }) : null
+			},
+			placeholder: localize('enterBackupUrl', "Please enter URL"),
+			ariaLabel: LocalizedStrings.BACKURL
+		};
+		const urlInputContainer = DOM.append(this._restoreFromUrlElement, DOM.$('.dialog-input-section'));
+		DOM.append(urlInputContainer, DOM.$('.dialog-label')).innerText = LocalizedStrings.BACKURL;
+
+		this._urlInputBox = new InputBox(DOM.append(urlInputContainer, DOM.$('.dialog-input')), this._contextViewService, urlValidationOptions);
+
+		const urlBrowseContainer = DOM.append(this._restoreFromUrlElement, DOM.$('.dialog-input-section'));
+		DOM.append(urlBrowseContainer, DOM.$('.dialog-label')).innerText = '';
+
+		let browseLabel = localize('restoreDialog.browse', "Browse");
+		this._browseUrlButton = new Button(DOM.append(urlBrowseContainer, DOM.$('.file-browser')), { secondary: true });
```

Comment:
Register (if you could add the register for other components in this dialog that didn't have it already that'd be great too)